src/core/Expander/ExpanderTitleButton/ExpanderTitleButton.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { Component, ReactNode } from 'react';
+import { default as styled } from 'styled-components';
+import classnames from 'classnames';
+import { withSuomifiDefaultProps } from '../../theme/utils';
+import { TokensProp, InternalTokensProp } from '../../theme';
+import { HtmlDiv, HtmlButton, HtmlButtonProps, HtmlSpan } from '../../../reset';
+import { expanderTitleButtonBaseStyles } from './ExpanderTitleButton.baseStyles';
+import { Icon } from '../../Icon/Icon';
+import { ExpanderConsumer, ExpanderTitleBaseProps } from '../Expander/Expander';
+import { VisuallyHidden } from '../../../components';
+
+const baseClassName = 'fi-expander_title-button';
+const titleOpenClassName = `${baseClassName}--open`;
+const titleButtonClassName = `${baseClassName}_button`;
+const iconClassName = `${baseClassName}-icon`;
+const iconOpenClassName = `${iconClassName}--open`;
+
+export interface ExpanderTitleButtonProps {
+  /** Custom classname to extend or customize */
+  className?: string;
+  /** Title for Expander */
+  children?: ReactNode;
+  /** Wrap the title button inside a heading tag */
+  asHeading?: 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
+  /** Additional text for closed expanders toggle button. E.g."open expander". */
+  ariaOpenText?: string;
+  /** Additional text for open expanders toggle button. E.g."close expander". */
+  ariaCloseText?: string;
+  /** Properties for title open/close toggle button */
+  toggleButtonProps?: Omit<
+    HtmlButtonProps,
+    | 'onClick'
+    | 'onMouseDown'
+    | 'onMouseUp'
+    | 'onKeyPress'
+    | 'onKeyUp'
+    | 'onKeyDown'
+  >;
+}
+
+interface InternalExpanderTitleButtonProps
+  extends ExpanderTitleButtonProps,
+    ExpanderTitleBaseProps {}
+
+class BaseExpanderTitleButton extends Component<
+  InternalExpanderTitleButtonProps
+> {
+  render() {
+    const {
+      ariaCloseText,
+      ariaOpenText,
+      asHeading,
+      children,
+      className,
+      toggleButtonProps,
+      consumer,
+      ...passProps
+    } = this.props;
+
+    return (
+      <HtmlDiv
+        {...passProps}
+        className={classnames(className, baseClassName, {
+          [titleOpenClassName]: !!consumer.open,
+        })}
+      >
+        <HtmlSpan {...(!!asHeading ? { as: asHeading } : {})}>
```

Comment:
Is it necessary to wrap to `span` when `asHeading` is not told?
Or was there some other benefit by doing like this?

Reply:
Not necessary, but would keep the wrapping structure uniform if done this way. I prefer not to use conditional wrapping structures with complex structures as the logic would be either in two places (both opening and closing tags) or the contents would have to be defined as const and have two completely separate blocks.